kentikapi/internal/api_payloads/saved_filter.go

```diff
@@ -0,0 +1,182 @@
+package api_payloads
+
+import (
+	"time"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/utils"
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+)
+
+type GetAllSavedFilterResponse []GetSavedFilterResponse
+
+func (p GetAllSavedFilterResponse) ToSavedFilters() ([]models.SavedFilter, error) {
+	var result []models.SavedFilter
+	err := utils.ConvertList(p, GetSavedFilterResponse.ToSavedFilter, &result)
+	if err != nil {
+		return nil, err
+	}
+	return result, err
+}
+
+type GetSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type CreateSavedFilterRequest struct {
+	SavedFilterPayload
+}
+
+type CreateSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type UpdateSavedFilterRequest struct {
+	SavedFilterPayload
+}
+
+type UpdateSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type SavedFilterPayload struct {
+	ID                IntAsString    `json:"id,omitempty" response:"get,post,put"`
```

Comment:
Response validation is performed on deserialized response and checks if the field is nil pointer/interface. If there is no e.g. filter_name in response, the field will have `""` value the validation will not detect that. This could be misleading. Because of that I propose to keep response validation only for pointer/interface types.

So for this payload it makes sense to have those tags only for CreatedDate and UpdatedDate.

Reply:
I see that ID and Company ID still have validation tags:
```
ID                IntAsString    `json:"id,omitempty" response:"get,post,put"`
CompanyID         models.ID      `json:"company_id,string,omitempty" response:"get,post,put"`
```

Aren't they redundant?

Reply:
It's a pity I cannot reopen the conversation BTW